### PR TITLE
Implement support for EFS

### LINKED ISSUE #311: Feature: Add ability to nuke EFS
Add the ability to nuke EFS

### PR DESCRIPTION
**Pull request by zackproser**
## Description

Fixes #311.

Implements support for EFS.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Implement support for inspecting and nuking Elastic FileSystems (efs).

### CODE REVIEW

File: aws/efs_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type TestElasticFileSystem struct {
+	Name *string
+	ID   *string
+}
+
+func createTestElasticFileSystem(t *testing.T, session *session.Session, name string) (*TestElasticFileSystem, error) {
```
Comment (denis256) [inline aws/efs_test.go:25]: Returned `error` is `nil` each time, maybe should be handled `err` from `cfg, err := awsconfig.LoadDefaultConfig`

**Review by denis256 — APPROVED** [review]

File: aws/efs_test.go (hunk 1 above)
Comment (zackproser) [reply aws/efs_test.go:25]: Thanks, Denis! Fixed in [efffae3](https://github.com/gruntwork-io/cloud-nuke/pull/353/commits/efffae353f0ba70afe35b337380bbc931ca467ba)

File: aws/efs.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		return []*string{}, errors.WithStackTrace(err)
+	}
+	svc := efs.NewFromConfig(cfg)
+
+	result, err := svc.DescribeFileSystems(context.TODO(), &efs.DescribeFileSystemsInput{})
+	if err != nil {
+		return []*string{}, errors.WithStackTrace(err)
+	}
+
+	Ids := []*string{}
```
[the remark this answers is not in the document]
Comment (zackproser) [reply aws/efs.go]: Fixed in [efffae3](https://github.com/gruntwork-io/cloud-nuke/pull/353/commits/efffae353f0ba70afe35b337380bbc931ca467ba)

File: aws/efs_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	cfg, err := awsconfig.LoadDefaultConfig(context.TODO(), awsconfig.WithRegion(aws.StringValue(session.Config.Region)))
+	if err != nil {
+		return errors.WithStackTrace(err)
```
Comment (denis256) [inline aws/efs_test.go:28]: I think should be returned 2 values to match function signature

Comment (zackproser) [reply aws/efs_test.go]: Thanks, fixed in ae94fc3

**Review by denis256 — APPROVED** [review]

**Review by rhoboat — APPROVED** [review]
LGTM, thanks for the review from Denis!

**Comment by zackproser** [comment]
Thanks for reviews Denis and Rho! Going to merge this in now.

**Pull request merged**